### PR TITLE
docs: clean up links

### PR DESCRIPTION
**Pull request by sjwang90**
clean up links from #10100

### CODE REVIEW

File: plugins/inputs/httpjson/README.md [hunk 1]
```diff
@@ -1,8 +1,8 @@
 # HTTP JSON Input Plugin
 
-The httpjson plugin collects data from HTTP URLs which respond with JSON.  It flattens the JSON and finds all numeric values, treating them as floats.
+### DEPRECATED in Telegraf v1.6: Use [HTTP input plugin](../http) as replacement.
```
Comment (srebhan) [inline plugins/inputs/httpjson/README.md:3]: How about
```suggestion
### DEPRECATED in Telegraf v1.6: Use the [http][] input plugin as replacement.
```
with adding the link to match the other link styles below?

**Review by srebhan — COMMENTED** [review]
Hey @sjwang90, thanks for unifying those. It now only leaves `http` to be different. Can we please also unify this one? Additionally, you need a resolve the conflicts...

**Comment by sjwang90** [comment]
thanks @srebhan! git confuses me sometimes. i think everything should be addressed.

**Review by srebhan — COMMENTED** [review]
Nice, thanks @sjwang90 for the update. Now that you touched the READMEs the new linter catches you and holds you responsible for fixing the mistakes of previous generations... ;-) Can you please take a look at the linter concerns?

File: plugins/inputs/jolokia/README.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # Jolokia Input Plugin
 
-### Deprecated in version 1.5: Please use the [jolokia2](https://github.com/influxdata/telegraf/tree/master/plugins/inputs/jolokia2) plugin.
+### Deprecated in version 1.5: Please use the [jolokia2][] plugin
```
Comment (powersj) [inline plugins/inputs/jolokia/README.md:3]: @sjwang90 - this should be `##` as headers should only increment or decrease by 1 level. As this is right after the top level it should only have two.

File: plugins/inputs/kafka_consumer_legacy/README.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # Kafka Consumer Legacy Input Plugin
 
-### Deprecated in version 1.4. Please use [Kafka Consumer input plugin](https://github.com/influxdata/telegraf/tree/master/plugins/inputs/kafka_consumer).
+### Deprecated in version 1.4. Please use [Kafka Consumer input plugin][]
```
Comment (powersj) [inline plugins/inputs/kafka_consumer_legacy/README.md:3]: Same here.

File: plugins/inputs/jolokia/README.md [hunk 4]
```diff
@@ -65,9 +65,11 @@
 
 The Jolokia plugin collects JVM metrics exposed as MBean's attributes through
 jolokia REST endpoint. All metrics are collected for each server configured.
+See [official Jolokia website](https://jolokia.org/) for more information.
 
-See: https://jolokia.org/
+# Measurements
```
Comment (powersj) [inline plugins/inputs/jolokia/README.md:70]: This should be '##' as well. as only the very top level should be '#'

File: plugins/inputs/jolokia/README.md (hunk 2 above)
Comment (powersj) [reply plugins/inputs/jolokia/README.md:3]:
```suggestion
## Deprecated in version 1.5: Please use the [jolokia2][] plugin
```

File: plugins/inputs/jolokia/README.md (hunk 4 above)
Comment (powersj) [reply plugins/inputs/jolokia/README.md:70]:
```suggestion
## Measurements
```

File: plugins/inputs/kafka_consumer_legacy/README.md (hunk 3 above)
Comment (powersj) [reply plugins/inputs/kafka_consumer_legacy/README.md:3]:
```suggestion
## Deprecated in version 1.4. Please use [Kafka Consumer input plugin][]
```

**Review by powersj — COMMENTED** [review]
Change the header levels in a couple of places and this is good to go.

**Review by powersj — APPROVED** [review]

**Pull request merged**